lib/trino-hive-formats/src/main/java/io/trino/hive/formats/avro/AvroTypeException.java

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.hive.formats.avro;
+
+import static java.util.Objects.requireNonNull;
+
+public class AvroTypeException
+        extends Exception
+{
+    public AvroTypeException(org.apache.avro.AvroTypeException runtimeAvroTypeException)
+    {
+        super(requireNonNull(runtimeAvroTypeException, "runtimeAvroTypeException is null").getMessage(), runtimeAvroTypeException.getCause());
```

Comment:
No need to check for null here -- we'll get a direct NPE and the reason will be obvious

Reply:
It seems better to pass `runtimeAvroTypeException` directly as the cause in order to preserve the correct stack trace. Otherwise, we won't know where the original was thrown from.